scalding-core/src/main/scala/com/twitter/scalding/serialization/JavaStreamEnrichments.scala

```diff
@@ -0,0 +1,228 @@
+    def readByte: Byte = readUnsignedByte.toByte
+
+    def readUnsignedByte: Int = {
+      // Note that Java, when you read a byte, returns a Int holding an unsigned byte.
+      // if the value is < 0, you hit EOF.
+      val c1 = s.read
+      if (c1 < 0) eof else c1
+    }
+    def readUnsignedShort: Int = {
+      val c1 = s.read
+      val c2 = s.read
+      if ((c1 | c2) < 0) eof else ((c1 << 8) | c2)
+    }
+
+    final def readFully(bytes: Array[Byte]): Unit = readFully(bytes, 0, bytes.length)
+
+    final def readFully(bytes: Array[Byte], offset: Int, len: Int): Unit = {
+      if (len < 0) throw new IndexOutOfBoundsException()
+
+      @annotation.tailrec
+      def go(o: Int, l: Int): Unit =
+        if (l == 0) ()
+        else {
+          val count = s.read(bytes, o, l)
+          if (count < 0) eof
+          else go(o + count, l - count)
+        }
+      go(offset, len)
+    }
+
+    def readDouble: Double = java.lang.Double.longBitsToDouble(readLong)
+    def readFloat: Float = java.lang.Float.intBitsToFloat(readInt)
+
+    def readInt: Int = {
+      val c1 = s.read
+      val c2 = s.read
+      val c3 = s.read
+      val c4 = s.read
+      // This is the algorithm from DataInputStream
+      if ((c1 | c2 | c3 | c4) < 0) eof else ((c1 << 24) | (c2 << 16) | (c3 << 8) | c4)
+    }
+    def readLong: Long = {
+      // This is the algorithm from DataInputStream
+      val buf = new Array[Byte](8)
+      readFully(buf)
+      (buf(0).toLong << 56) +
+        ((buf(1) & 255).toLong << 48) +
+        ((buf(2) & 255).toLong << 40) +
+        ((buf(3) & 255).toLong << 32) +
+        ((buf(4) & 255).toLong << 24) +
+        ((buf(5) & 255) << 16) +
+        ((buf(6) & 255) << 8) +
+        (buf(7) & 255)
+    }
+
+    def readShort: Short = {
+      val c1 = s.read
+      val c2 = s.read
+      // This is the algorithm from DataInputStream
+      if ((c1 | c2) < 0) eof else ((c1 << 8) | c2).toShort
+    }
+
+    /**
+     * This reads a varInt encoding that only encodes non-negative
+     * numbers. It uses:
+     * 1 byte for values 0 - 255,
+     * 3 bytes for 256 - 65535,
+     * 7 bytes for 65536 - Int.MaxValue
+     */
+    final def readSize: Int = {
+      val c1 = readUnsignedByte
+      if (c1 < ((1 << 8) - 1)) c1
+      else {
+        val c2 = readUnsignedShort
+        if (c2 < ((1 << 16) - 1)) c2
+        else readInt
+      }
+    }
+
+    final def skipFully(count: Long): Unit = {
+      @annotation.tailrec
+      def go(c: Long): Unit = {
+        val skipped = s.skip(c)
+        if (skipped == c) ()
+        else if (skipped == 0L) throw new Exception(s"count, c, skipped = ${(count, c, skipped)}")
+        else go(c - skipped)
+      }
+      if (count != 0L) go(count) else ()
+    }
+  }
+
+  implicit class RichOutputStream(val s: OutputStream) extends AnyVal {
+    def writeBoolean(b: Boolean): Unit = if (b) s.write(1: Byte) else s.write(0: Byte)
+
+    def writeBytes(b: Array[Byte], off: Int, len: Int): Unit = {
```

Comment:
what's this for? does the caller already have s and can call this on s directly?